src/FSharp.Stats/Fitting/GeneralisedLinearModel.fs

```diff
@@ -0,0 +1,697 @@
+/// </summary>
+type LinkFunction =
+    {
+        /// Gets the link function.
+        getLink: float -> float
+        /// Gets the inverse link function.
+        getInvLink: float -> float
+        /// Gets the derivative of the link function.
+        getDeriv: float -> float
+        /// Gets the derivative of the inverse link function.
+        getInvLinkDerivative: float -> float
+    }
+
+/// <summary>
+/// Represents the return type of a Generalised Linear Model (GLM).
+/// </summary>
+/// <remarks>
+/// This type contains the following elements:
+/// <list type="bullet">
+/// <item>
+/// <description>
+/// <c>mX</c>: The coefficients used in the GLM.
+/// </description>
+/// </item>
+/// <item>
+/// <description>
+/// <c>mu</c>: The predicted mean values of the GLM.
+/// </description>
+/// </item>
+/// </list>
+/// </remarks>
+type GLMReturn = 
+    {
+        /// The coefficients used in the GLM.
+        mX: Vector<float>
+        /// The predicted mean values of the GLM.
+        mu: Vector<float>
+    }
+
+/// <summary>
+/// Represents the statistics of a Generalised Linear Model (GLM).
+/// </summary>
+/// <remarks>
+/// This type contains the following elements:
+/// <list type="bullet">
+/// <item>
+/// <description>
+/// <c>LogLikelihood</c>: The log-likelihood of the GLM.
+/// </description>
+/// </item>
+/// <item>
+/// <description>
+/// <c>Deviance</c>: The deviance of the GLM.
+/// </description>
+/// </item>
+/// <item>
+/// <description>
+/// <c>PearsonChi2</c>: The Pearson chi-squared statistic of the GLM.
+/// </description>
+/// </item>
+/// </list>
+/// </remarks>
+type GLMStatisticsModel = 
+    {
+        /// The log-likelihood of the GLM.
+        LogLikelihood: float
+        /// The deviance of the GLM.
+        Deviance: float
+        /// The Pearson chi-squared statistic of the GLM.
+        PearsonChi2: float
+        //PseudoR2:float
+    }
+
+/// <summary>
+/// Represents the parameters of a Generalised Linear Model (GLM).
+/// </summary>
+/// <remarks>
+/// This type contains the following elements:
+/// <list type="bullet">
+/// <item>
+/// <description>
+/// <c>Coefficient</c>: The coefficient of the parameter.
+/// </description>
+/// </item>
+/// <item>
+/// <description>
+/// <c>StandardError</c>: The standard error of the parameter.
+/// </description>
+/// </item>
+/// <item>
+/// <description>
+/// <c>ZScore</c>: The Z-score of the parameter.
+/// </description>
+/// </item>
+/// <item>
+/// <description>
+/// <c>PersonOfZ</c>: The person of Z of the parameter.
+/// </description>
+/// </item>
+/// </list>
+/// </remarks>
+type GLMStatisticsPrameter = 
+    {
+        //Name:string
+        /// The coefficient of the parameter.
+        Coefficient: float
+        /// The standard error of the parameter.
+        StandardError: float
+        /// The Z-score of the parameter.
+        ZScore: float
+        /// The person of Z of the parameter.
+        PersonOfZ: float
```

Comment:
Is this field actually called PersonOfZ? I did no research but it seems to me as if standardError and ZScore would be associated to a p-value?

Reply:
You are right, this is supposed to be Pearson. I fixed this.